BLSQ/iaso
# IA-2565: Org units search change source should reset parent org unit

If source field isn't actively cleared before switching source, then parent selection doesn't take into account the source switch
In Org unit search, if source field isn't actively cleared before switching source, then parent selection doesn't take into account the source switch (see attached video).

https://github.com/BLSQ/iaso/assets/12494624/e04a1149-0cf2-4cbe-aec2-a55d455b4ad4



Related JIRA tickets : IA-2565

## Self proofreading checklist


## Changes

- fix typings issues
- delete levels in search if source is changed

## How to test

Make a string search leading to few results on a specific source. (take a print screen to compare results)
Change the source and choose a  parent org unit. and keep the same string search and search.
List should be empty or different.
Switch back to first source, keep search string and search again, results should be the same as the print screen.


## Print screen / video


https://github.com/BLSQ/iaso/assets/12494624/228a1d4b-4d64-44c3-ad40-3a592c5488b6



## Notes

Cypress test are not implemented here, we should keep in mind those race conditions and test it !
